mutiny-core/src/node.rs

```diff
@@ -166,55 +166,176 @@ impl PubkeyConnectionInfo {
     }
 }
 
-pub(crate) struct Node<S: MutinyStorage> {
-    pub _uuid: String,
-    pub child_index: u32,
-    stopped_components: Arc<RwLock<Vec<bool>>>,
-    pub pubkey: PublicKey,
-    pub peer_manager: Arc<PeerManagerImpl<S>>,
-    pub keys_manager: Arc<PhantomKeysManager<S>>,
-    pub channel_manager: Arc<PhantomChannelManager<S>>,
-    pub chain_monitor: Arc<ChainMonitor<S>>,
-    pub fee_estimator: Arc<MutinyFeeEstimator<S>>,
-    network: Network,
-    pub persister: Arc<MutinyNodePersister<S>>,
-    wallet: Arc<OnChainWallet<S>>,
-    pub(crate) logger: Arc<MutinyLogger>,
-    pub(crate) lsp_client: Option<AnyLsp<S>>,
-    pub(crate) sync_lock: Arc<Mutex<()>>,
-    stop: Arc<AtomicBool>,
+pub struct NodeBuilder<S: MutinyStorage> {
+    // required
+    xprivkey: ExtendedPrivKey,
+    storage: S,
+    uuid: Option<String>,
+    node_index: Option<NodeIndex>,
+    gossip_sync: Option<Arc<RapidGossipSync>>,
+    scorer: Option<Arc<utils::Mutex<HubPreferentialScorer>>>,
+    chain: Option<Arc<MutinyChain<S>>>,
+    fee_estimator: Option<Arc<MutinyFeeEstimator<S>>>,
+    wallet: Option<Arc<OnChainWallet<S>>>,
+    esplora: Option<Arc<AsyncClient>>,
     #[cfg(target_arch = "wasm32")]
-    websocket_proxy_addr: String,
+    websocket_proxy_addr: Option<String>,
+
+    // optional
+    lsp_config: Option<LspConfig>,
+    logger: Option<Arc<MutinyLogger>>,
+    network: Network,
```

Comment:
i think we should require network, can definitely see some tricky bugs coming because someone forgot to set the network somewhere